src/renderer/backend/chat.ts

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * Finds basic info, like contact id and the related chat ID of a contact based
+ * on it's email address.
+ *
+ * The contact id can be null if this email address is unknown. If no chat
+ * exists yet with this contact the chat id will be null as well.
+ */
+export async function getChatInfoByEmail(
+  accountId: number,
+  email: string
+): Promise<{
+  chatId: number | null
+  contactId: number | null
+}> {
+  const contactId = await BackendRemote.rpc.lookupContactIdByAddr(
+    accountId,
+    email
+  )
+
+  const chatId = contactId
+    ? await BackendRemote.rpc.getChatIdByContactId(accountId, contactId)
+    : null
+
+  return {
+    contactId,
+    chatId,
+  }
+}
+
+export async function saveLastChatId(accountId: number, chatId: number) {
+  await BackendRemote.rpc.setConfig(accountId, 'ui.lastchatid', `${chatId}`)
+}
+
+export async function getLastChatId(accountId: number): Promise<number | null> {
+  const chatId = await BackendRemote.rpc.getConfig(accountId, 'ui.lastchatid')
+
+  if (typeof chatId === 'string') {
+    return parseInt(chatId, 10)
+  }
+
+  return null
+}
+
+export async function muteChat(
+  accountId: number,
+  chatId: number,
+  duration: T.MuteDuration
+) {
+  await BackendRemote.rpc.setChatMuteDuration(accountId, chatId, duration)
+}
+
+export async function unmuteChat(accountId: number, chatId: number) {
+  await BackendRemote.rpc.setChatMuteDuration(accountId, chatId, {
+    kind: 'NotMuted',
+  })
+}
+
+export function markChatAsSeen(accountId: number, chatId: number) {
+  // Mark all messages in chat as "seen" in core backend
+  BackendRemote.rpc.marknoticedChat(accountId, chatId)
+  debouncedUpdateBadgeCounter()
+
+  // Remove potential system notifications for this chat
+  clearNotificationsForChat(accountId, chatId)
+}
+
+export async function createChatByContactId(
+  accountId: number,
+  contactId: number | null,
+  email?: string
+): Promise<number> {
+  // Create contact first with given email address if it doesn't exist yet
+  if (!contactId) {
+    if (!email) {
+      throw new Error('either contactId or email needs to be set')
+    }
+
+    contactId = await BackendRemote.rpc.createContact(accountId, email, null)
+  }
+
+  return await BackendRemote.rpc.createChatByContactId(accountId, contactId)
+}
+
+/**
+ * Returns true if all contacts of a given list are verified, otherwise false.
+ */
+export async function areAllContactsVerified(
+  accountId: number,
+  contactIds: number[]
+): Promise<boolean> {
+  const contacts = await BackendRemote.rpc.getContactsByIds(
+    accountId,
+    contactIds
+  )
+
+  return !contactIds.some(contactId => {
+    return !contacts[contactId].isVerified
+  })
+}
+
+/**
+ * Helper method to determine the chat id of the "Device Messages" read-only chat.
+ *
+ * Note that there's currently no API to retrieve this id from the backend, this
+ * is why we're iterating over all currently available chats instead.
+ */
```

Comment:
sidenote, I know the comment is not new:

I just tested performance and came to the conclusion that we do not need make a core api as long as we only use it on fresh accounts it's fast enough.

my account main account: ~163ms
a new account that only has like 20 chats: ~9-11ms
completely fresh account: ~2-3ms

This method is not exactly efficient, but we don't use it enough to justify moving it to core.

Reply:
But the disadvantage to it is that it triggers multiple renders instead of just one... anyways I tried both ways and can not see a huge difference on my fast device.